src/store/getters/state-getters.ts

```diff
@@ -190,6 +194,26 @@ export const isBusySaving = (state: any): boolean => {
   return (state.stateModel.isSaving || state.stateModel.isSavingResuming || state.stateModel.isFilingPaying)
 }
 
+/** Is true when the step is valid. */
+export const isDefineCompanyValid = (state: any): boolean => {
+  return state.stateModel.defineCompanyStep.valid
+}
+
+/** Is true when the step is valid. */
+export const isAddPeopleAndRolesValid = (state: any): boolean => {
+  return state.stateModel.addPeopleAndRoleStep.valid
+}
+
+/** Is true when the step is valid. */
+export const isCreateShareStructureValid = (state: any): boolean => {
+  return state.stateModel.createShareStructureStep.valid
+}
+
+/** Is true when the step is valid. */
+export const isIncorporationAgreementValid = (state: any): boolean => {
+  return state.stateModel.incorporationAgreementStep.valid
```

Comment:
Can we change the store IF to this?

```
stateModel: {
   validFlags: {
      incorporationAgreementStep: boolean,
      createShareStructureSte: boolean,
      etc
   }
}
```

... In order to get away from "step-based data" and move towards "component-based data". Plus it will be easier to extend with more components.

Reply:
For sure, lets do this now.

Reply:
On second thought, we should do this in the framework ticket. We will be refactoring the store for this exact purpose and this change falls perfectly in with that.

Reply:
You're right. I agree!